Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/file_bugs.py

```diff
@@ -0,0 +1,197 @@
+                sys.stderr.write('Failed to create new issue\n')
+                return
+            print("Created '{}'".format(issue))
+            if radar:
+                print('Found existing rdar to CC: ' + str(radar))
+                issue.cc_radar(block=True, radar=radar)
+        if needs_radar:
+            radar = issue.cc_radar(block=True, radar=None)
+            print("Created '{}'".format(radar))
+
+        # Try to mutate the commit message and insert our newly filed issue.
+        # Currently, we can only do this for the HEAD command, and if there's no staged
+        # changes that our amend would accidentally bring in. Is this a sufficient check?
+        # Detached HEAD state? Error checking on the amend git command?
+        # Should we pre-warn at the start of the command if we won't be able to write
+        # the results?
+        if repository.find('HEAD') != commit:
+            print('Commit is not head, cannot edit to insert issue')
+            return
+
+        if repository.modified(staged=True) != []:
+            print('Staged files found, cannot edit to insert issue')
+            return
+
+        modified = commit.message.replace(cls.BUGZILLA_PLACEHOLDER, issue.link)
+        if radar:
+            modified = modified.replace(cls.RADAR_PLACEHOLDER, radar.link)
+
+        print('Writing new commit message:' + modified)
+        run([repository.executable(), 'commit', '--amend', '-m', modified])
+
+        # Can we run a `git rebase --exec 'git webkit rewrite-message' <range>` style
+        # command?
+        # We'd need to store all our new messages somewhere that the other script can
+        # access them. Maybe the env? Can't index by commit hash, since the rebase operation
+        # will be changing them.
+        # rewrite-message subcommand can lookup the new message (somehow) and then git
+        # commit --amend -m <new-message>
+
+    @classmethod
+    def create_bugs(cls, repository, args, branch_point):
+        commits = list(repository.commits(begin=dict(hash=branch_point.hash), end=dict(branch=repository.branch)))
+
+        for commit in commits:
+            cls.create_bug(repository, commit)
+
+        # Should we try print a summary of filed issues that we couldn't write into commit
+        # messages, so that it's easy for the user to then use them in an interactive rebase.
+        return 0
+
+    @classmethod
+    def main(cls, args, repository, **kwargs):
+        if not isinstance(repository, local.Git):
+            sys.stderr.write("Can only '{}' on a native Git repository\n".format(cls.name))
+            return 1
+
+        # pull_request_branch_point expects an `issue` variable that it uses as part of its
+        # logic for choosing the branch point. Given that not having an issue is the whole
+        # reason we're running this code, it might be a sign that this logic isn't going to
+        # be correct for what we want.
+        args.issue = None
```

Comment:
Maybe we should just make `PullRequest.pull_request_branch_point` work without the issue argument than force the caller to pass in an empty one. It should work with `issue = None`, if I recall what we're doing in that function.